From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gregor MacLennan <gmaclennan@gmail.com>
Date: Tue, 28 Jul 2026 19:20:28 +0000
Subject: [PATCH 08/19] src: skip unclonable env vars on Android
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

node_env_var.cc: continue (not abort) when an env var like LD_PRELOAD can't be cloned.

Co-authored-by: Andre Staltz <andre@staltz.com>
Co-authored-by: Julia Samól <j.samol@papers.ch>
Co-authored-by: Jaime Bernardo <jaime@janeasystems.com>
---
 src/node_env_var.cc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/node_env_var.cc b/src/node_env_var.cc
index 5550a4bee3..761fc10b23 100644
--- a/src/node_env_var.cc
+++ b/src/node_env_var.cc
@@ -235,7 +235,9 @@ std::shared_ptr<KVStore> KVStore::Clone(Isolate* isolate) const {
     }
     CHECK(key->IsString());
     if (!Get(isolate, key.As<String>()).ToLocal(&value)) {
-      return nullptr;
+      // On Android, it's possible that some environment variables get removed,
+      // such as LD_PRELOAD. Skip them rather than aborting the whole clone.
+      continue;
     }
     copy->Set(isolate, key.As<String>(), value.As<String>());
   }
